### ISSUE TITLE
h() consumes functions-as-children if not wrapped in array

### ISSUE THREAD

**Issue by krainboltgreene** [snabbdom/snabbdom#344]
http://requirebin.com/?gist=2173039ac453c9f0225a47f53bc26041

**Comment by caridy** [comment]
> `h("li", {}, () => "c"),`
> `h("li", {}, [() => "c"]),`

the second one "sort-of" work because of the nature of JS in which the function passed as the first element of that array happens to be an object (all functions are objects), but as a general rule, we want snabbdom to be less forgiving about what you pass as part of the vnode in favor of letting other higher order layers to take care of that normalization process.

Remember that you can implement your own `h` function that does all sort of normalization to produce valid vnodes.

**Comment by krainboltgreene** [comment]
Yes, and actually I currently handle this bug in snabbdom-helpwrs (by
arrayifying it)

**Comment by mightyiam** [comment]
I changed the title to be more general. I'm not sure whether `h` should be more strict about children only in TypeScript types or at runtime, as well. At runtime there's a performance penalty for that, whether meaningful or not.
